src/compiler/parser.ts

```diff
@@ -4616,10 +4631,10 @@ namespace ts {
                 factory.createConditionalExpression(
                     leftOperand,
                     questionToken,
-                    doOutsideOfContext(disallowInAndDecoratorContext, parseAssignmentExpressionOrHigher),
+                    doOutsideOfContext(disallowInAndDecoratorContext, () => parseAssignmentExpressionOrHigher(/*parsingConditionalExpression*/ true)),
                     colonToken = parseExpectedToken(SyntaxKind.ColonToken),
                     nodeIsPresent(colonToken)
-                        ? parseAssignmentExpressionOrHigher()
+                        ? parseAssignmentExpressionOrHigher(/*parsingConditionalExpression*/ true)

```

Comment:
I agree, I think a context flag would be easier to read and possibly faster. These are currently the only two places that are needed, right?

Actually, why are they both `true`? Where is `false` ever provided?

Edit: Oh, the parameter is optional. Yeah, I'd much prefer a context flag to a lot of identical optional parameters.

Reply:
I was worried you'd say that, darn.

I guess my main annoyance was how many "okay, it's time to not be in this context" now calls I would have to add to all of the calls that `parseAssignmentExpressionOrHigher` (and all of the function it calls), but that's no better than the `for in` thing I'm already not a fan of...